src/igraph/__init__.py

```diff
@@ -2814,12 +2814,32 @@ def Incidence(klass, *args, **kwds):
           what the value is. If C{True}, non-zero entries are rounded up to
           the nearest integer and this will be the number of multiple edges
           created.
+        @param weighted: defines whether to create a weighted graph from the
+          incidence matrix. If it is c{None} then an unweighted graph is created
+          and the multiple argument is used to determine the edges of the graph.
+          If it is a character constant then for every non-zero matrix entry, an
+          edge is created and the value of the entry is added as an edge attribute
+          named by the weighted argument. If it is C{True} then a weighted graph
+          is created and the name of the edge attribute will be ‘weight’.
 
         @return: the graph with a binary vertex attribute named C{"type"} that
           stores the vertex classes.
         """
+        weighted = kwds.pop("weighted", False)
+        if weighted:
+          kwds["multiple"] = False
         result, types = klass._Incidence(*args, **kwds)
         result.vs["type"] = types
+        if weighted:
+            weight_attr = "weight" if weighted == True else weighted
+            mat = args[0]
+            _, rows, columns = result.get_incidence()
+            for edge in result.es:
+                source, target = edge.tuple
+                if source in rows:
+                    edge[weight_attr] = mat[source][target - len(rows)]
```

Comment:
Lift `len(rows)` to a variable outside the loop (say `num_vertices_of_first_kind`) and just use that variable inside the loop to avoid calling `len()` all the time.